tests/unit/specification/parser_test.py

```diff
@@ -2802,3 +2803,317 @@ def test_parse_reserved_word_as_channel_name() -> None:
+            then Length
+            if Option_Type = OT_Data;
+         Length : Length
+            then Data
+            with Size => Length * 8;
+        Data : Opaque;
+      end message;
+   type Options is sequence of Option;
+   type Message is
+      message
+         Message_Type : Message_Type
+            then null
+            if Message_Type = MT_Null
+            then Data
+            with Size => Message'Last - Message_Type'Last
+            if Message_Type = MT_Unconstrained_Data
+            then Length
+            if Message_Type /= MT_Null
+               and Message_Type /= MT_Unconstrained_Data
+               and Message_Type /= MT_Unconstrained_Options
+            then Options
+            with Size => Message'Last - Message_Type'Last
+            if Message_Type = MT_Unconstrained_Options;
+         Length : Length
+            then Data
+            with Size => Length * 8
+            if Message_Type = MT_Data
+            then Option_Types
+            with Size => Length * 8
+            if Message_Type = MT_Option_Types
+            then Options
+            with Size => Length * 8
+            if Message_Type = MT_Options
+            then Value
+            if Message_Type = MT_Value
+               and Length = 1
+            then Values
+            with Size => Length * 8
+            if Message_Type = MT_Values;
+         Data : Opaque
+            then null;
+         Option_Types : Option_Types
+            then null;
+         Options : Options
+            then null;
+         Value : ValueT
+            then null;
+         Values : Values;
+      end message;
+
+   for Message use (Data => Option);
+
+   generic
+       Channel : Channel with Readable, Writable;
+   session Session with
+       Initial => Start,
+       Final => Terminated
+   is
+       Msg : Message;
+   begin
+      state Start is
+      begin
+         Channel'Read (Msg);
+      transition
+         goto Reply
+            if Msg'Valid = True
+            and Msg.Message_Type = MT_Data
+            and Msg.Length = 1
+         goto Terminated
+      end Start;
+      state Reply is
+      begin
+         Msg := Message'(Message_Type => MT_Data, Length => 1, Data => [2]);
+      transition
+         goto Msg_Write
+      exception
+         goto Terminated
+      end Reply;
+      state Msg_Write is
+      begin
+         Channel'Write (Msg);
+      transition
+         goto Next
+      end Msg_Write;
+      state Next is
+         Msg2 : Message;
+      begin
+         Channel'Read(Msg2);
+      transition
+         goto Terminated
+      end Next;
+      state Terminated is null state;
+   end Session;
+   end Test;
+"""
+    test_spec = tmp_path / "test.rflx"
+    test_rfi = tmp_path / "test.rfi"
+    test_spec.write_text(content, encoding="utf-8")
+    test_rfi.write_text(rfi_content)
+    regex = re.compile("test.rfi:.*" + match_error, re.DOTALL)
+    if raises:
+        with pytest.raises(RecordFluxError, match=regex):
```

Comment:
The error message contains the full filename including a temp path, so are you suggesting that I should construct the full error message regex with something like that:
```
   regex = re.compile("^" + tmp_path + "test.rfi: + ...)
```
That would work but it's really cumbersome. Also, the error messages from pydantic are multiline, so this will be even more cumbersome.
We could also remove some of those tests as they are effectively testing pydantic (don't contribute to code coverage). But then it would be nice to have these tests in case we change the implementation (use something other than pydantic).
I think what I really would like to have is a framework where I can update the expected outputs automatically. We can discuss this during our call today.

Reply:
I think my regexes now would detect the case of additional error messages, since d095c573a1defa433803ca5da716b75140cbca52